library/src/main/java/com/novoda/downloadmanager/CompletedDownloadBatchesExtractor.java

```diff
@@ -90,39 +87,36 @@
                             fileIds.add(originalFileId);
                         }
 
-                        if (originalFileId == null) {
-                            newBatchBuilder.downloadFrom(originalNetworkAddress)
-                                    .apply();
-                        } else {
-                            newBatchBuilder.downloadFrom(originalNetworkAddress)
-                                    .withIdentifier(DownloadFileIdCreator.createFrom(originalFileId))
-                                    .apply();
-                        }
-
                         String rawNewFilePath = new LiteFilePath(sanitizedOriginalUniqueFileLocation).path();
                         FilePath newFilePath = MigrationPathExtractor.extractMigrationPath(basePath, rawNewFilePath, downloadBatchId);
 
-                        FilePath originalFilePath = new LiteFilePath(originalFileLocation);
-                        long rawFileSize = filePersistence.getCurrentSize(originalFilePath);
+                        long rawFileSize = fileSizeExtractor.fileSizeFor(originalFileLocation);
```

Comment:
`fileSizeExtractor` was created to avoid breaking the tests so that we can mock the dependency.